### DISCUSSION TITLE
Frequent wifi disconnection

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#562] [Q&A] [answered]
Hello,

Due to my poor wifi at home, my Birdnet-Pi sometimes disconnects from my wifi (e.g I have to relaunch my router which is another raspberry pi) and it doesn't auto reconnect to the wifi. Do you know any method to make my Birdnet-pi auto reconnects to the wifi network when this happens? Currently I can only hard reboot the Birdnet-Pi unplugging from power socket, which is bad.

Thank you.

**Comment by lloydbayley** [comment]
Woah...that's really weird. Mine reconnects to the Wifi if I happen to power-cycle (accidentally unplug) the wifi. Is there anything special about your Wifi setup on the Pi? Did you hard-code any extras?

**Reply by blogdemoi** [reply]
`Is there anything special about your Wifi setup on the Pi? Did you hard-code any extras?`

No, nothing special. In the mean time I just ordered  an Ethernet cable RJ45 so I can connect my Birdnet-Pi directly to the Pi.

**Reply by blogdemoi** [reply]
I will also try to add this script using cron: https://gist.github.com/jacobpretorius/1c479ec38e8021196215fd2e9ae74675

**Reply by lloydbayley** [reply]
Whilst that might be a stop-gap solution, there has to be a reason that it keeps disconnecting. It's very weird.
So rebooting the birdnetpi doesn't work, you have to physically power it down?

**Reply by blogdemoi** [reply]
> So rebooting the birdnetpi doesn't work, you have to physically power it down?

Yes, I can't even ssh into.

**Reply by lloydbayley** [reply]
Hmm...okay, I'd be putting a monitor and keyboard on it to see what state things are in when it drops out. 
Anyway, try the power management thing, who knows.
FYI, mine has power management still turned on and it reconnects fine but it's certainly worth a try.

**Reply by lloydbayley** [reply]
oh ok, this has been marked as answered. Sorry. As you were.

**Reply by blogdemoi** [reply]
Thank you for trying to help me. I will try first mcguirepr89 solution, then the cron solution then the RJ45 cable... Now I wait for the next disconnection.

**Comment by mcguirepr89** [comment, marked as answer]
`wlan0` on a Raspberry Pi has power management automatically turned on. I would recommend disabling this on boot to address your issue.

Just above `exit 0` in `/etc/rc.local`, add this line:
```
iwconfig wlan0 power off
```

and that will disable power management (the thing that cuts the power to your wireless network card after it loses the connection to your WiFi, preventing it from reconnecting in the future).

**Reply by blogdemoi** [reply]
Thank you, I will try this first.
